### PR TITLE
Handle batch calls

### PR DESCRIPTION
**Pull request by m4gus88**
Addressing issue #411, added a handler for batch calls.

### CODE REVIEW

**Comment by fsouza** [comment]
Hi @m4gus88, thanks for contributing! Can you add some tests?

**Comment by m4gus88** [comment]
@fsouza Any chance this will be merged and released any time soon?

**Comment by fsouza** [comment]
> @fsouza Any chance this will be merged and released any time soon?

Hey, somehow I missed this after the last approval for running, I'm really sorry about that. I'll review it today and if everything looks good I'll merge it and release it!

File: fakestorage/server.go [hunk 1]
```diff
@@ -291,6 +301,71 @@ func (s *Server) Client() *storage.Client {
...
+			http.Error(partResponseWriter, "unable to process request", http.StatusBadRequest)
+			writeMultipartResponse(partResponseWriter.Result(), partWriter, contentId)
+			continue
+		}
+
+		s.mux.ServeHTTP(partResponseWriter, partRequest)
+		writeMultipartResponse(partResponseWriter.Result(), partWriter, contentId)
+	}
+
+	mw.Close()
```
Comment (fsouza) [inline fakestorage/server.go:357]: can you move this to line 315, with a `defer`?

File: fakestorage/server_test.go [hunk 2]
```diff
@@ -13,6 +13,8 @@ import (
 	"strings"
 	"testing"
 
+	"google.golang.org/api/iterator"
+
```
Comment (fsouza) [inline fakestorage/server_test.go:17]: can you delete this line? (I realize it's something goimports add, but it's annoying lol)

File: fakestorage/server.go [hunk 3]
```diff
@@ -291,6 +301,71 @@ func (s *Server) Client() *storage.Client {
...
+	w.Header().Set("Content-Type", "multipart/mixed; boundary="+mw.Boundary())
+
+	w.WriteHeader(http.StatusOK)
+	part, err := reader.NextPart()
+	for ; err == nil; part, err = reader.NextPart() {
+		contentId := part.Header.Get("Content-ID")
+		if contentId == "" {
+			// missing content ID, skip
+			continue
+		}
```
Comment (fsouza) [inline fakestorage/server.go:321]: nit: can you rename this to `contentID`?

```suggestion
		contentID := part.Header.Get("Content-ID")
		if contentID == "" {
			// missing content ID, skip
			continue
		}
```

**Review by fsouza — APPROVED** [review]
Looks good! Just some minor points, but I'll merge it once they're addressed!

File: fakestorage/server_test.go [hunk 4]
```diff
@@ -13,8 +13,6 @@ import (
 	"strings"
 	"testing"
 
-	"google.golang.org/api/iterator"
-
```
Comment (fsouza) [inline fakestorage/server_test.go]: oh I meant deleting just the blank line, the import is necessary.

Comment (m4gus88) [reply fakestorage/server_test.go]: ah, alright, I suspected something was going wrong there :D

**Review by fsouza — APPROVED** [review]

**Comment by fsouza** [comment]
Thank you!

**Pull request merged**